Add Web of Science Reviewer Recognition Service plugin for OJS 3.4

Alex-nz opened the pull request:
Add Web of Science Reviewer Recognition Service plugin for OJS 3.4 into the plugin gallery. This plugin is the heavily rebranded old Publons plugin, hence I put these together in the file.

Alex-nz commented:
Hi PKP team, can we please get this PR reviewed?

bozana commented on plugins.xml:
```diff
@@ -9768,6 +9768,41 @@
 			<description>Minor improvements for Portuguese translations.</description>
 		</release>
 	</plugin>
+	<plugin category="generic" product="webOfScience">
```
Hi @Alex-nz, thanks a lot for the contribution!
I went through the plugin code and would have a few comments/questions:
- The handler https://github.com/clarivate/wos_reviewer_recognition_plugin_ojs_3/blob/main/WOSHandler.php is missing the authorization function. This way everybody could call the function `exportReview`. I believe that `addRoleAssigment` should be used, so that only reviewers (an user with a reviewer role) can call the function, and it should also be checked if the request context corresponds to the submission context (having the submission from the reviewAssignment) -- so that a reviewer from one journal cannot call that function for a review in another journal (that maybe has no plugin enabled) -- e.g. if the parameter `reviewId`, that the function expects, would be manipulated. I believe the WOS service would surely not work correctly for example in case the user is not a reviewer any more, or for a review in another journal, but the safety should be ensured by coding it.
- When are these lines the case: https://github.com/clarivate/wos_reviewer_recognition_plugin_ojs_3/blob/main/WOSHandler.php#L191 and https://github.com/clarivate/wos_reviewer_recognition_plugin_ojs_3/blob/main/WOSHandler.php#L214 -- when and how would `$_SERVER["HTTP_WOS_URL"]` be set? 
- Instead of using php curl you should rather use Guzzle -- you can get the HTTP client from the `Application` class. This client is properly configured for proxies if the server is behind one.
- I think `<certification type="reviewed"/>` should be used below.
Please let me know if you would have any questions and what do you think about the suggestions...
Thanks a lot!

Alex-nz replied:
Thanks for the review @bozana. I would have to defer answering your comments to the developer who worked on this plugin. @v0k1c when you are back from vacation, can you please look into this?

Alex-nz commented:
Hi @bozana , we've made the requested updates and updated the PR, can you please re-review? 

> I think <certification type="reviewed"/> should be used below.

^ I've used the same certification type that we were given when published the plugin previously, under Publons name. I have updated it to "reviewed" as you suggested.

bozana commented:
Hi @Alex-nz, 
I would have two more comments: 
- I think in the function exportReview it should be also checked that the $request->getContext()->getId() is the same as the submission->getData('contextId')
- Why are you using https://fonts.googleapis.com/css?family=Roboto? -- We have chosen the fonts under consideration to support wide range of languages etc. They are downloaded and provided with OJS. So plugins would preferably use them (you would not need to do anything for that). 
This way, using 'https://fonts.googleapis.com/css?family=Roboto' would provide user information (IP) address to google (which would be good to inform/ask the user about). Thus, if you would really need that CSS, maybe to download it and provide it within the plugin, but I believe there should be no need to use anything extra.

And finally, let me double check regarding the certification level with the team...

Thanks a lot!

Alex-nz commented:
thank you @bozana , @v0k1c and I are looking into your comments.

Alex-nz commented:
@bozana @v0k1c has made the changes as requested and I've updated this PR to reflect the new release. Can you please re-review?

v0k1c commented:
Hi @bozana! I've just double checked, and Open Sans (although available within the theme plugin?) is not loaded by default? All I'm seeing that is loaded is Noto Sans and Fontawesome. Would it be safe to import/enable it through theme plugin, or what is the preferred way to use additional fonts that are available there? Asking because I can't find anything related to this within the available documentation.

bozana commented:
Hi @Alex-nz and @v0k1c, I will take a new look at the code, and will forward your question about the fonts to @jardakotesovec :-)
Thanks a lot!

bozana commented:
Hi @Alex-nz and @v0k1c, the code now looks good, no further comments :-)
I also double checked with the team that certification level = "reviewed" is good.
Lets just wait what @jardakotesovec says about the fonts and then we can merge.
Thanks a lot for the great work/contribution!

Alex-nz commented:
@bozana , @jardakotesovec, could we please get the direction for how to proceed with fonts here? We have customers waiting to start using the plugin.

jardakotesovec commented:
@v0k1c My understanding is that this plugin renders UI only on editorial side of OJS, after reviewer completes the review, right?

On Editorial site we use noto-sans, I would recommend just to use that, so we don't increase the resources that needs to be fetched unnecessarily. 

Themes utilise various fonts, but these are specific only for 'reader side'. On editorial side we don't intend provide flexibility on theming or use of different fonts.

Does that answer the question?

Alex-nz commented:
thanks @jardakotesovec , yes, you have answered the question.
@v0k1c has updated the plugin accordingly 
@bozana can you please re-review and merge if all well?

bozana commented:
Thanks a lot all! Merged :tada:

Alex-nz commented:
Great, thank you very much everyone involved!

Alex-nz commented:
@bozana can you please confirm when is this plugin going to be available within the gallery? I have a local installation of OJS 3.4.0.4 which I used for testing the plugin, but I don't see it in the plugin gallery list.

bozana commented:
Hi @Alex-nz, I can see it now in my local 3.4 installation... Do you see it now?

Alex-nz commented:
@bozana Yes, I can see it today, but for some reason it wasn't present there yesterday. Anyways, it's all good then, thank you.

bozana commented:
Hi @Alex-nz and @v0k1c, may I ask you here a question -- I cannot find anything online:
We are thinking how to integrate OJS with ResearcherID: Would it be possible for OJS users to, similarly to ORCID, add their authenticated/verified (and not just manually entered) ResearcherID to their OJS profile? -- I cannot find any API endpoint that would allow this kind of authentication and retrieval of user's Web of Science ResearcherID. Could you maybe help me?
And, if we would work on other kind of integration with Clarivate products, what would be best way to ask question, would it be possible to get test accounts, etc.
If you want you can also answer me per e-mail: bozana dot bokan at posteo dot net.
I would be very happy about any further information. 
Thanks a lot!

Alex-nz commented:
hi @bozana, sorry for a delayed response: I was on vacation and only returned today. I've emailed you.

bozana commented:
Thanks a lot @Alex-nz!
